run_diff_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wptdashboard
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const mockTestPath = "/mock/path.html"
+
+func TestDiffResults_NoDifference(t *testing.T) {
+	assertNoDeltaDifferences(t, []int {0, 1}, []int {0, 1})
+	assertNoDeltaDifferences(t, []int {3, 4}, []int {3, 4})
+}
+
+func TestDiffResults_Difference(t *testing.T) {
+	// One test now passing
+	assertDelta(t, []int {0, 1}, []int {1, 1}, []int {1, 1})
+
+	// One test now failing
+	assertDelta(t, []int {1, 1}, []int {0, 1}, []int {1, 1})
+
+	// Two tests, one now failing
+	assertDelta(t, []int {2, 2}, []int {1, 2}, []int {1, 2})
+
+	// Three tests, two now passing
+	assertDelta(t, []int {1, 3}, []int {3, 3}, []int {2, 3})
+}
+
+func TestDiffResults_Added(t *testing.T) {
+	// One new test, all passing
+	assertDelta(t, []int {1, 1}, []int {2, 2}, []int {1, 2})
+
+	// One new test, all failing
+	assertDelta(t, []int {0, 1}, []int {0, 2}, []int {1, 2})
+
+	// One new test, new test passing
+	assertDelta(t, []int {0, 1}, []int {1, 2}, []int {1, 2})
```

Comment:
I I'm reading this right, the numbers returned as the same when a new test is pass and when it's failing, is that right? That would make it impossible to notify about new failing tests but not new passing tests, which I think we should.

Reply:
Yeah, currently the case. Removing the abs call would let you see negative values; however, this API will be extended to take a filters param, and you'd filter to only newly-failing in the fetch which is used for notifying.

Reply:
Sounds like a plan!